src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresDirectoryDAO.scala

```diff
@@ -542,6 +542,25 @@ class PostgresDirectoryDAO(protected val writeDbRef: DbReference, protected val
     })
   }
 
+  override def acceptTermsOfService(userId: WorkbenchUserId, tosVersion: String, samRequestContext: SamRequestContext): IO[Boolean] = {
+    serializableWriteTransaction("acceptTermsOfService", samRequestContext)({ implicit session =>
+      val u = UserTable.column
+      samsql"""update ${UserTable.table} set ${u.acceptedTosVersion} = ${tosVersion}
+              where ${u.id} = ${userId}
+              and (${u.acceptedTosVersion} is null
```

Comment:
Why are we checking this? Wouldn't it be safe to just set `acceptedTosVersion` to whatever this is called with? Is the thinking just that we might as well surface whether the ToS acceptance actually changed or not, even if it's not consumed upstream at the moment?

Reply:
I was keeping somewhat to the original behavior. Tests were using it. I also prefer to return something other than Unit.